tests/helpers_tests/test_event_tag_utils.py

```diff
@@ -11,70 +11,132 @@
+        self.assertIsNone(event_tag_utils.get_revenue_value({'revenue': [1, 2, 3]}, logger=logger.SimpleLogger()))
+        self.assertIsNone(event_tag_utils.get_revenue_value({'revenue': {'a', 'b', 'c'}}, logger=logger.SimpleLogger()))
+    mock_logger.assert_called_with(
+        enums.LogLevels.WARNING,
+        'Revenue value is not an integer or float, or is not a numeric string.'
+    )
+    self.assertEqual(3, mock_logger.call_count)
+
+  def test_get_revenue_value__revenue_tag_invalid_string(self):
+    """ Test that revenue value is not returned when revenue event tag has invalid string value. """
+    with mock.patch('optimizely.logger.SimpleLogger.log') as mock_logger:
+        self.assertIsNone(event_tag_utils.get_revenue_value({'revenue': 'string'}, logger=logger.SimpleLogger()))
+    mock_logger.assert_called_once_with(enums.LogLevels.WARNING, 'Revenue value is not a numeric string.')
+
+  def test_get_revenue_value__revenue_tag_invalid_numeric(self):
+    """ Test that revenue value is not returned when revenue event tag has invalid numeric value """
+    with mock.patch('optimizely.logger.SimpleLogger.log') as mock_logger:
+        self.assertIsNone(event_tag_utils.get_revenue_value({'revenue': 0.5}, logger=logger.SimpleLogger()))
+        self.assertIsNone(event_tag_utils.get_revenue_value({'revenue': '0.5'}, logger=logger.SimpleLogger()))
+    mock_logger.assert_called_with(enums.LogLevels.WARNING, 'Failed to parse revenue value "0.5" from event tags.')
+    self.assertEqual(2, mock_logger.call_count)
+
+  def test_get_revenue_value__revenue_tag_valid_numeric(self):
+    """ Test that correct revenue value is returned when revenue event tag has valid numeric value """
+    with mock.patch('optimizely.logger.SimpleLogger.log') as mock_logger:
+        self.assertEqual(65536, event_tag_utils.get_revenue_value({'revenue': 65536.0}, logger=logger.SimpleLogger()))
+        self.assertEqual(65536, event_tag_utils.get_revenue_value({'revenue': '65536.0'}, logger=logger.SimpleLogger()))
+    mock_logger.assert_called_with(enums.LogLevels.INFO, 'Parsed revenue value "65536.0" from event tags.')
+    self.assertEqual(2, mock_logger.call_count)
+
+    with mock.patch('optimizely.logger.SimpleLogger.log') as mock_logger:
+        self.assertEqual(65536, event_tag_utils.get_revenue_value({'revenue': 65536}, logger=logger.SimpleLogger()))
+    mock_logger.assert_called_once_with(enums.LogLevels.INFO, 'Parsed revenue value "65536" from event tags.')
+
+    with mock.patch('optimizely.logger.SimpleLogger.log') as mock_logger:
+        self.assertEqual(
+            9223372036854775807,
+            event_tag_utils.get_revenue_value({'revenue': 9223372036854775807}, logger=logger.SimpleLogger())
+        )
+    mock_logger.assert_called_once_with(
+        enums.LogLevels.INFO,
+        'Parsed revenue value "9223372036854775807" from event tags.'
+    )
+
+  def test_get_revenue_value__revenue_tag_zero(self):
```

Comment:
Why a separate test for 0 value? Looks like 0 is a special test case. You can merge these into above valid revenue tag method